lib/spack/spack/util/web.py

```diff
@@ -516,6 +544,9 @@ def _urlopen(req, *args, **kwargs):
     if url_util.parse(url).scheme == 's3':
         import spack.s3_handler
         opener = spack.s3_handler.open
+    elif url_util.parse(url).scheme == 'gs':
+        import spack.gcs_handler
+        return spack.gcs_handler.gcs_open(req)
```

Comment:
I'm not sure what the `*args` and `**kwargs` could be in this context, but it's probably a good idea to pass those along to your opener.

```suggestion
    elif url_util.parse(url).scheme == 'gs':
        import spack.gcs_handler
        opener = spack.gcs_handler.gcs_open
```

... and then leave the calling to the code immediately following?